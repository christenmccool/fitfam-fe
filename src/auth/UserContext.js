import React from 'react';

/** Context: provide access to user object throughout app */

const UserContext = React.createContext();

export default UserContext;
