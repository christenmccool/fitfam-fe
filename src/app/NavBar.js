import React, {useContext} from 'react';
import { Link as RouterLink } from 'react-router-dom';

import Box from '@mui/material/Box';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import FitnessCenterIcon from '@mui/icons-material/FitnessCenter';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import Avatar from '@mui/material/Avatar';


import UserContext from '../auth/UserContext';

//NavBar Component for FitFam app
const NavBar = ({logout}) => {
  const {user} = useContext(UserContext);

  return (
    <AppBar position="static">
      <Toolbar>
        <Box sx={{ flexGrow: 1, display: 'flex' }}>
          <IconButton 
            aria-label="go to FitFam homepage" 
            component={RouterLink} 
            to="/"
            color="inherit" 
          >
            <FitnessCenterIcon fontSize="large" />
          </IconButton>
          <Button 
            component={RouterLink} 
            to="/"
            color="inherit"
            size="large"
            sx={{p: 1, textTransform: 'none'}}
          >
            <Typography variant="h5">
              FitFam
            </Typography>
          </Button>
        </Box>
        <Box>
          {user ?
            <Box sx={{display: 'flex'}}>
              <Avatar 
                component={RouterLink} 
                to="/profile"
                src={user.imageUrl}
                sx={{bgcolor: 'white.main', color: 'primary.main', mr: 2}}
              >
              </Avatar>
              <Button 
                component={RouterLink} 
                to="/"
                color="inherit"
                onClick={logout}
                variant="outlined"
                sx={{py: '2px'}}
              >
                Logout
              </Button>
            </Box>
            :
            <Box>
              <Button 
                color="primary"
                component={RouterLink}
                to={`/login`}
                variant="contained"
                color="white"
                sx={{
                  p: '2px', 
                  m: 1, 
                  color: 'primary.main'
                }}
              >
                Login
              </Button>
              <Button 
                color="inherit"
                component={RouterLink}
                to={`/signup`}
                variant="outlined"
                sx={{p: '2px'}}
              >
                Signup
              </Button>
            </Box>
          }
        </Box>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;