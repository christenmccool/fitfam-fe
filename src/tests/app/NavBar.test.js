import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from '@mui/material/styles';

import { theme, demoUser } from "../testutils";

import NavBar from "../../app/NavBar";
import UserContext from "../../auth/UserContext";


it('renders without crashing when not logged in', () => {
  render(
    <MemoryRouter>
      <ThemeProvider theme={theme}>
        <UserContext.Provider value={{user: null}}>
          <NavBar />
        </UserContext.Provider>
      </ThemeProvider>
    </MemoryRouter>
  );
});


it("matches snapshot when not logged in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <ThemeProvider theme={theme}>
        <UserContext.Provider value={{user: null}}>
          <NavBar />
        </UserContext.Provider>
      </ThemeProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("matches snapshot when logged in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <ThemeProvider theme={theme}>
        <UserContext.Provider value={{user: demoUser}}>
          <NavBar />
        </UserContext.Provider>
      </ThemeProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders Fitfam, Login, and Signup links when not logged in", function () {                
  const { queryByText } = render(
    <MemoryRouter>
      <ThemeProvider theme={theme}>
        <UserContext.Provider value={{user: null}}>
          <NavBar />
        </UserContext.Provider>
      </ThemeProvider>
    </MemoryRouter>
  );
  expect(queryByText("FitFam")).toBeInTheDocument();
  expect(queryByText("Login")).toBeInTheDocument();
  expect(queryByText("Signup")).toBeInTheDocument();
});


it("renders Fitfam and Logout links when not logged in", function () {                
  const { queryByText } = render(
    <MemoryRouter>
      <ThemeProvider theme={theme}>
        <UserContext.Provider value={{user: demoUser}}>
          <NavBar />
        </UserContext.Provider>
      </ThemeProvider>
    </MemoryRouter>
  );
  expect(queryByText("FitFam")).toBeInTheDocument();
  expect(queryByText("Logout")).toBeInTheDocument();
});
