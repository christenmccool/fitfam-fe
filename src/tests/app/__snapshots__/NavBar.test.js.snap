// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-11ouz13-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-1l0r181"
      >
        <a
          aria-label="go to FitFam homepage"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          href="/"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="FitnessCenterIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-colorInherit MuiButtonBase-root  css-1pbqo9m-MuiButtonBase-root-MuiButton-root"
          href="/"
          tabindex="0"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-9e04yy-MuiTypography-root"
          >
            FitFam
          </h5>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <a
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-fxb010-MuiAvatar-root"
            href="/profile"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </a>
          <a
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-73cyr3-MuiButtonBase-root-MuiButton-root"
            href="/"
            tabindex="0"
          >
            Logout
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`matches snapshot when not logged in 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-11ouz13-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    >
      <div
        class="MuiBox-root css-1l0r181"
      >
        <a
          aria-label="go to FitFam homepage"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          href="/"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="FitnessCenterIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <a
          class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-colorInherit MuiButtonBase-root  css-1pbqo9m-MuiButtonBase-root-MuiButton-root"
          href="/"
          tabindex="0"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-9e04yy-MuiTypography-root"
          >
            FitFam
          </h5>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
        >
          <a
            class="MuiButton-root MuiButton-contained MuiButton-containedWhite MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-144b0kh-MuiButtonBase-root-MuiButton-root"
            href="/login"
            tabindex="0"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-15jxpu6-MuiButtonBase-root-MuiButton-root"
            href="/signup"
            tabindex="0"
          >
            Signup
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
